### PR TITLE
Upgrade vercel.json

### PR DESCRIPTION
**Pull request by colebemis**
Our Vercel configuration was using deprecated settings. This PR updates `vercel.json` to use the most up-to-date configuration settings.

[Vercel configuration docs](https://vercel.com/docs/configuration)

### CODE REVIEW

File: now.json [hunk 1]
```diff
@@ -1,28 +1,20 @@
...
-  "name": "primer-style",
-  "alias": "primer.style",
-  "builds": [
-    {
-      "src": "package.json",
-      "use": "@now/static-build",
-      "config": {"distDir": "public"}
-    }
+  "rewrites": [
+    {"source": "/components(/.*)?", "destination": "https://primer-components.vercel.app"},
```
Comment (colebemis) [inline now.json:3]:
```suggestion
    {"source": "/components/:path*", "destination": "https://primer-components.vercel.app/:path*"},
```

File: now.json [hunk 2]
```diff
@@ -1,28 +1,20 @@
...
-  "name": "primer-style",
-  "alias": "primer.style",
-  "builds": [
-    {
-      "src": "package.json",
-      "use": "@now/static-build",
-      "config": {"distDir": "public"}
-    }
+  "rewrites": [
+    {"source": "/components/:path*", "destination": "https://primer-components.vercel.app/:path*"},
```
Comment (colebemis) [inline now.json:3]:
```suggestion
    {"source": "/components", "destination": "https://primer-components.vercel.app"},
```

File: now.json [hunk 3]
```diff
@@ -1,28 +1,20 @@
...
-  "name": "primer-style",
-  "alias": "primer.style",
-  "builds": [
-    {
-      "src": "package.json",
-      "use": "@now/static-build",
-      "config": {"distDir": "public"}
-    }
+  "rewrites": [
+    {"source": "/components", "destination": "https://primer-components.vercel.app"},
```
Comment (colebemis) [inline now.json:3]:
```suggestion
    {"source": "/components/:path*", "destination": "https://primer-components.vercel.app"},
```

File: now.json [hunk 4]
```diff
@@ -1,28 +1,20 @@
...
-  "name": "primer-style",
-  "alias": "primer.style",
-  "builds": [
-    {
-      "src": "package.json",
-      "use": "@now/static-build",
-      "config": {"distDir": "public"}
-    }
+  "rewrites": [
+    {"source": "/components/:path*", "destination": "https://primer-components.vercel.app"},
```
Comment (colebemis) [inline now.json:3]:
```suggestion
    {"source": "/components(/.*)?", "destination": "https://primer-components.vercel.app"},
```

**Review by jonrohan — APPROVED** [review]
I'll approve but have no idea if this is correct. 👍🏻

**Comment by colebemis** [comment]
@jonrohan The redirects in the deploy previews are working so I think it should work in production.

**Pull request merged**